collections/_tutorials/2022-09-12-tutorial-cvae.md

```diff
@@ -0,0 +1,296 @@
+
+$$\mathcal{L} = \int{q(z/x,y) \cdot \left[log\left(p(y/x,z)\right) + log\left(\frac{p(z/x)}{q(z/x,y)}\right) \right] \,dz}$$
+
+$$\mathcal{L} = \int{q(z/x,y) \cdot log\left(p(y/x,z)\right) \,dz} \,+\, \int{q(z/x,y) \cdot log\left(\frac{p(z/x)}{q(z/x,y)}\right) \,dz}$$
+
+$$\mathcal{L} =  \mathbb{E}_{z\sim q(z/x,y)} \left[log\left(p(y/x,z)\right)\right] - D_{KL}\left(q(z/x,y)\parallel p(z/x)\right)$$
+
+where $$\mathbb{E}_{z\sim q(z/x,y)}$$ is the mathematical expectation with respect to $$q(z/x,y)$$. 
+
+&nbsp;
+
+At this stage of analysis, it is necessary to model $$p(y/x,y)$$ via an ***a priori distribution***. Let's first note that $$p(y/x,z)$$ is modeled by a neural network $$f(\cdot)$$ so that $$\hat{y}=f(x,z)$$. Since this function is deterministic, it will allow to model $$p\left(y/\hat{y}\right)$$. By approximating $$p\left(y/\hat{y}\right)$$ by a Bernoulli distribution, we have
+
+$$\mathbb{E}_{z\sim q(z/x,y)} \left[log\left(p(y/\hat{y})\right)\right] = \mathbb{E}_{z\sim q(z/x,y)} \left[log\left( {\hat{y}}^y \cdot \left(1-\hat{y}\right)^{1-y} \right)\right]$$
+
+$$ = \mathbb{E}_{z\sim q(z/x,y)} \left[ y \, log\left( \hat{y} \right) + \left(1-y\right) \, log\left(1-\hat{y}\right) \right]$$
+
+$$ = \mathbb{E}_{z\sim q(z/x,y)} \left[-CE\left( y,f\left(x,z\right)\right)\right]$$
+
+where $$CE(\cdot)$$ corresponds to the conventional ***cross entropy function*** !
+
+&nbsp;
+
+Thanks to this modeling, we are finally looking for:
+
+$$\left(f^*,g^*,h^*,k^*,l^*\right) = \underset{(f,g,h,k,l)}{\arg\max} \,\,\, \left( \mathbb{E}_{z\sim q(z/x,y)} [-CE\left( y,f\left(x,z\right)\right)] - D_{KL}(\underbrace{q(z/x,y)}_{g,h}\parallel \underbrace{p(z/x)}_{k,l}) \right)$$
+
+$$\left(f^*,g^*,h^*,k^*,l^*\right) = \underset{(f,g,h,k,l)}{\arg\min} \,\,\, \left( \mathbb{E}_{z\sim q(z/x,y)} [CE\left( y,f\left(x,z\right)\right)] + D_{KL}(\underbrace{q(z/x,y)}_{g,h}\parallel \underbrace{p(z/x)}_{k,l}) \right)$$
+
+&nbsp;
+
+## Various scenarios
+
+There are different exploitations of the cVAE formalism depending on the prior modeling $$p(z/x)$$ and the content of the $$y$$ references.
+
+### Pior modeling
+
+* The prior $$p(z/x)$$ outputs a latent variable $$z$$ depending on the input $$x$$. This means that the corresponding latent space will be structured according to a varying input $$x$$ as illustrated in the figure below.
+
+![](/collections/images/cvae/cvae_prior_depending_on_x.jpg)
+
+
+* Several works in the literature propose to relax this constraint to make the latent variables statistically independent of input variables, i.e. $$p(z/x) = p(z)$$ with $$z \sim \mathcal{N}\left(0,I\right)$$. This implies that the latent space is forced to be centered at the origin with unit variance, which makes the posterior modeling strategy close to that used in standard VAE, as illustrated in the figure below.
+
+![](/collections/images/cvae/cvae_prior_no_depending_on_x.jpg)
+
+### References content
```

Comment:
I'm not sure about this title. When I first read it, I didn't intuitively understand what _content_ was supposed to mean. Maybe something like _Nature of the references_? But I'm not too confident about my own suggestion, so if you have any other idea than go for it!

Reply:
Thanks for the suggestion, I kept it ;)